open-api/rest-catalog-open-api.yaml

```diff
@@ -1435,12 +1463,102 @@ components:
           additionalProperties:
             type: string
 
+    OAuthTokenRequest:
+      description:
+        OAuth2 client credentials or token exchange request
+
+
+        See -
+
+        - Client credentials flow - https://datatracker.ietf.org/doc/html/rfc6749#section-4.4
+
+        - Token exchange flow - https://datatracker.ietf.org/doc/html/rfc8693
+      type: object
+      required:
+        - grant_type
+      properties:
+        grant_type:
+          type: string
+          enum:
+            - client_credentials
+            - urn:ietf:params:oauth:grant-type:token-exchange
+        client_id:
+          type: string
+          description:
+            Client ID for client credentials flow.
+
+
+            This can be sent in the request body, but OAuth2 recomments sending it in
+            a Basic Authorization header.
+        client_secret:
+          type: string
+          description:
+            Client secret for client credentials flow.
+
+
+            This can be sent in the request body, but OAuth2 recomments sending it in
+            a Basic Authorization header.
+        subject_token:
+          type: string
+          description: Subject token for token exchange request
+        subject_token_type:
+          type: string
+          description: Subject token type for token exchange request
+        scope:
+          type: string
 
   #############################
   # Reusable Response Objects #
   #############################
   responses:
 
+    OAuthTokenResponse:
```

Comment:
This doesn't quite match the [spec](https://datatracker.ietf.org/doc/html/rfc6749#section-5.1) for granting a token (it looks like it matches the token exchange response though)